src/ngraph/runtime/cpu/pass/cpu_mat_fusion.cpp

```diff
@@ -327,21 +327,35 @@ bool runtime::cpu::pass::CPURnnMatFusion::run_on_function(std::shared_ptr<Functi
             auto add_node = std::make_shared<op::Add>(dot_node, bias_broadcast_node);
             const auto& add_shape = add_node->get_shape();
 
+            // we will sort the captured Add(Dot(X, W) + B) as per the the slice ordering of X
+            // this will simplify the replace_node logic
+            auto lambda = [&](const std::shared_ptr<Node> node1,
```

Comment:
A bit nitpicking, could you please change `lambda` to something like `compare_slice`.